api/C/include/csi_ll.h

```diff
@@ -36,5 +38,37 @@ typedef enum {
  */
 unsigned csi_ll_set_interrupt_enables(unsigned int_enables);
 
+/*
+ * Set the frequency of the system timer.  Note that there is typically a single
+ * timer for all harts in the system, so this function can affect the operation of
+ * other harts.  When using the RVM-CSI high-level interrupt and timer support
+ * module, the timer will be compared against a compare register for each hart in
+ * order to produce a regular timer interrupt at a tick rate configured using
+ * csi_set_timer_tick, which is used for timing purposes.  This function must run
+ * in machine mode.
+ *
+ * @param timer_freq_mhz: System timer frequency in MHz
+ * @return : Status of operation.  CSI_ERROR will be returned if the request is
+ * invalid.
+ */
+csi_status_t csi_timer_config(unsigned timer_freq_mhz);
+
+/*
+ * Read the current timer value.  This function can be called from M-mode or
+ * U-mode.  However, on systems where the timer is not directly readable from
+ * U-mode, the function will have to ECALL to M-mode to make the read, which will
+ * likely make the value innaccurate due to the delay incurred by this.
+ *
+ * @return : Current timer value
+ */
+long long csi_read_mtime(void);
```

Comment:
I remember there is also a `time` register which can be accessed in u-mode, maybe we can use read time to get time in u   mode, see https://github.com/riscv/riscv-isa-manual/blob/main/src/priv-csrs.adoc#csr-listing

and `long long` could be changed to `unsigned long long` or `uint64_t` since time should be above zero

Reply:
I thought `time` csr is just a mirror copy for `mtime` csr in u-mode, so user mode application can read time value via `rdtime` instruction. But if the API is `csi_read_mtime`, then I think it is ok to use directly `mtime`